src/core/net/mdns.cpp

```diff
@@ -2948,6 +2985,18 @@ void Core::TxMessage::AppendServicesDnssdName(Section aSection)
     return;
 }
 
+void Core::TxMessage::AddQuestionFrom(OwnedPtr<Message> &aMessage)
+{
+    uint16_t offset = sizeof(Header);
+
+    IgnoreError(Name(*aMessage, sizeof(Header)).AppendTo(*mMsgPtr));
+    IgnoreError(Name::ParseName(*aMessage, offset));
+    // RFC6762, section 6.7:
+    // Legacy Unicast Response must repeat the question
+    IgnoreError(mMsgPtr->AppendBytesFromMessage(*aMessage, offset, sizeof(ot::Dns::Question)));
+    IncrementRecordCount(kQuestionSection);
```

Comment:
Can consider coping the entire question (including name) using `AppendBytesFromMessage`.
How about something like this? 

```suggestion
    IgnoreError(Name::ParseName(*aMessage, offset));
    offset += sizeof(ot::Dns::Question);
    SuccessOrAssert(mMsgPtr->AppendBytesFromMessage(*aMessage, sizeof(Header), offset - sizeof(Header));
    IncrementRecordCount(kQuestionSection);
```

Reply:
Updated the code based on your proposal. Thanks for the feedback!